gravitee-apim-e2e/api-test/src/portal/portal-api-view-and-search.spec.ts

```diff
@@ -0,0 +1,174 @@
+const configurationApiAsAdmin = new ConfigurationApi(forManagementAsAdminUser());
+
+async function createAndPublish(apiManagementClient: APIsApi, attributes?: Partial<UpdateApiEntity>): Promise<ApiEntity> {
+  let createdApi = await apiManagementClient.createApi({
+    orgId,
+    envId,
+    newApiEntity: ApisFaker.newApi(),
+  });
+  await apiManagementClient.updateApi({
+    orgId,
+    envId,
+    api: createdApi.id,
+    updateApiEntity: UpdateApiEntityFromJSON({ ...createdApi, lifecycle_state: ApiLifecycleState.PUBLISHED, ...attributes }),
+  });
+  return createdApi;
+}
+
+describe('Portal - View and search APIs', () => {
+  let apiWith1Label: ApiEntity;
+  let apiWith2Labels: ApiEntity;
+  let apiFeatured: ApiEntity;
+  let apiWithCategory: ApiEntity;
+  let createdCategory: CategoryEntity;
+
+  beforeAll(async () => {
+    // create all APIs needed for testing
+    createdCategory = await succeed(
+      configurationApiAsAdmin.createCategoryRaw({
+        orgId,
+        envId,
+        newCategoryEntity: { name: 'cat1' },
+      }),
+    );
+    apiWithCategory = await createAndPublish(apisManagementApiAsAdmin, { categories: ['cat1'], description: 'API with one category' });
+    apiWith1Label = await createAndPublish(apisManagementApiAsAdmin, { labels: ['testlabel1'], description: 'API with one label' });
+    apiWith2Labels = await createAndPublish(apisManagementApiAsAdmin, {
+      labels: ['testlabel1', 'testlabel2'],
+      description: 'API with two labels',
+    });
+    apiFeatured = await createAndPublish(apisManagementApiAsAdmin, { description: 'Featured API' });
+    await succeed(configurationApiAsAdmin.createTopApiRaw({ orgId, envId, newTopApiEntity: { api: apiFeatured.id } }));
+  });
+
+  afterAll(async () => {
+    // delete all APIs that were created for testing
+    await apisManagementApiAsAdmin.deleteApi({ orgId, envId, api: apiWith1Label.id });
+    await apisManagementApiAsAdmin.deleteApi({ orgId, envId, api: apiWith2Labels.id });
+    await apisManagementApiAsAdmin.deleteApi({ orgId, envId, api: apiFeatured.id });
+    await apisManagementApiAsAdmin.deleteApi({ orgId, envId, api: apiWithCategory.id });
+    await configurationApiAsAdmin.deleteCategory({ orgId, envId, categoryId: createdCategory.id });
+  });
+
+  describe('View APIs in the APIM portal', function () {
+    describe('Filter API list regarding FilterApiQuery', () => {
+      test('should list all published APIs', async () => {
+        const getApisResponse = await succeed(apiPortalApiAsAdmin.getApisRaw({ filter: FilterApiQuery.ALL }));
+        const filteredApis = getApisResponse.data;
+        expect(filteredApis).toHaveLength(4);
+        expect(filteredApis.some((filteredApi) => filteredApi.id === apiWith1Label.id)).toBeTruthy();
+        expect(filteredApis.some((filteredApi) => filteredApi.id === apiWith2Labels.id)).toBeTruthy();
+        expect(filteredApis.some((filteredApi) => filteredApi.id === apiFeatured.id)).toBeTruthy();
+      });
```

Comment:
is it done on purpose to not test `apiWithCategory` ?

Reply:
Oh, good point, I missed that! Thx